controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageclassclaim
+
+import (
+	"context"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// StorageClassClaimReconciler reconciles a StorageClassClaim object
+// nolint
+type StorageClassClaimReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+
+func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := ctrllog.FromContext(ctx)
+	log = log.WithValues("StorageClassClaim", request)
+	log.Info("Reconciling StorageClassClaim.")
+
+	// Fetch the StorageClassClaim instance
+	claim := &ocsv1alpha1.StorageClassClaim{}
+	if err := r.Client.Get(ctx, request.NamespacedName, claim); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("StorageClassClaim resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to get StorageClassClaim.")
+		return reconcile.Result{}, err
+	}
+
```

Comment:
Can we add the status update pattern, the same way we have on storagecluster controller, storageconsumer controller, and other operators we wrote? 

You can create an empty `reconcilePhases` function to facilitate that code